src/ui/auth/authenticator.go

```diff
@@ -55,9 +55,13 @@ type AuthenticateHelper interface {
 	// put the id in the pointer of user model, if it does exist, fill in the user model based
 	// on the data record of the user
 	OnBoardUser(u *models.User) error
+	// Create a group in harbor DB
+	OnBoardGroup(g *models.UserGroup, groupName string) error
 	// Get user information from account repository
 	SearchUser(username string) (*models.User, error)
-	// Update user information after authenticate, such as Onboard or sync info etc
+	// Search a group based on specific authentication
+	SearchGroup(groupDN string) (*models.UserGroup, error)
```

Comment:
should this be groupName or filter?  If it's a filter, shall we make this an interface or struct to make it extensible?